lib/index.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @module lifion-verify-deps
+ */
+
+'use strict';
+
+const chalk = require('chalk');
+const path = require('path');
+const semver = require('semver');
+const { exec } = require('child_process');
+const { promisify } = require('util');
+
+const execAsync = promisify(exec);
+
+async function getLatestVersions(name) {
+  const { stdout } = await execAsync(`npm view ${name} versions --json`);
+  try {
+    return JSON.parse(stdout);
+  } catch (err) {
+    throw new Error(`Failed to parse output from NPM view - ${err.toString()}`);
+  }
+}
+
+async function getLatestVersion(name, wanted) {
+  const versions = await getLatestVersions(name);
+  const applicableVersions = versions.filter(i => semver.satisfies(i, wanted));
+  applicableVersions.sort((a, b) => semver.rcompare(a, b));
+  return applicableVersions[0];
+}
+
+function getInstalledVersion(currentDir, name) {
+  try {
+    return require(path.join(currentDir, 'node_modules', name, 'package.json')).version;
+  } catch (err) {
+    return null;
+  }
+}
+
+function pushPkgs({ dir, logger, deps = {}, type, pkgs }) {
+  return Object.keys(deps).map(async name => {
+    let wanted = deps[name];
+    if (!wanted.startsWith('^')) wanted = `^${wanted}`;
+    const installed = getInstalledVersion(dir, name);
+    const latest = await getLatestVersion(name, wanted);
+    const wantedFixed = wanted.substr(1);
+    const shouldBeInstalled =
+      installed === null || wantedFixed !== installed || installed !== latest;
+    if (shouldBeInstalled) {
+      const warning =
+        installed !== null
+          ? `outdated: ${chalk.red(
+              wantedFixed !== installed ? wantedFixed : installed
+            )} → ${chalk.green(latest)}`
+          : chalk.red('not installed');
+      logger.info(`${chalk.red(name)} is ${warning}`);
+    }
+    pkgs.push({ name, wanted, installed, type, latest, shouldBeInstalled });
+  });
+}
+
+function getPkgIds(filteredPkgs) {
+  return filteredPkgs.map(({ name, latest }) => `${name}@${latest}`).join(' ');
+}
+
+/**
+ * Verifies the dependencies listed in the package.json of the given directory.
+ * @param {object} params - Contains the directory path and optional custom logger.
+ */
+
+async function verifyDeps({ dir, logger = console }) {
```

Comment:
Default `params` to empty object. Otherwise calling `verifyDeps` without arguments will throw an exception.